header-validator/src/validate-json.ts

```diff
@@ -413,54 +561,77 @@ export function validateSource(ctx: Context, source: Json): void {
   }
 }
 
-const aggregatableTriggerData = list((ctx, value) =>
-  validate(ctx, value, {
-    key_piece: required(hex128),
-    source_keys: optional(
-      list(string(unique()), { maxLength: limits.maxAggregationKeys })
-    ),
-    ...filterFields,
-  })
-)
+function sourceKeys(ctx: Context, j: Json): Set<string> | undefined {
+  return set(ctx, j, string, { maxLength: limits.maxAggregationKeys })
+}
+
+function aggregatableTriggerData(ctx: Context, j: Json): void[] | undefined {
```

Comment:
Eventually instead of `void[]` these functions will be returning things like `AggregatableTriggerDatum[]`, where `AggregatableTriggerDatum` is a class composed of individually values corresponding to the validated fields. This is a temporary situation.

Reply:
sounds good.